Mathlib/Topology/Instances/NNReal/Lemmas.lean

```diff
@@ -232,4 +232,46 @@ theorem tendsto_of_antitone {f : ℕ → ℝ≥0} (h_ant : Antitone f) :
 
 end Monotone
 
+theorem iSup_pow {ι : Sort*} [Nonempty ι] {f : ι → ℝ≥0} (hf_bdd : BddAbove (range f)) (n : ℕ) :
+    (⨆ i : ι, f i) ^ n = ⨆ i : ι, f i ^ n := by
+  by_cases hn : n = 0
+  · simp [hn]
+  · simpa using (NNReal.powOrderIso n hn).map_ciSup hf_bdd
+
 end NNReal
+
+open NNReal in
+theorem Real.iSup_pow {ι : Sort*} [Nonempty ι] {f : ι → ℝ} (hf_bdd : BddAbove (range f))
```

Comment:
Oh you have dropped the n!=0 criterion? Earlier I suggested dropping the Nonempty \io criterion but you could drop the n!=0 criterion instead if you keep the nonempty one, right?

Reply:
Okay, I've found what went wrong in my reasoning. Will push a fix soon